components/identities/actions/autoRenewalModal.tsx

```diff
@@ -0,0 +1,268 @@
+import { Modal } from "@mui/material";
+import {
+  useAccount,
+  useContractRead,
+  useContractWrite,
+  useTransactionManager,
+} from "@starknet-react/core";
+import React, { FunctionComponent, useEffect, useState } from "react";
+import { usePricingContract } from "../../../hooks/contracts";
+import styles from "../../../styles/components/autoRenewal.module.css";
+import Button from "../../UI/button";
+import { timestampToReadableDate } from "../../../utils/dateService";
+import { Abi, Call } from "starknet";
+import ConfirmationTx from "../../UI/confirmationTx";
+import UsForm from "../../domains/usForm";
+import salesTax from "sales-tax";
+import TextField from "../../UI/textField";
+import {
+  computeMetadataHash,
+  generateSalt,
+} from "../../../utils/userDataService";
+import registrationCalls from "../../../utils/callData/registrationCalls";
+import { isValidEmail } from "../../../utils/stringService";
+import { applyRateToBigInt, gweiToEth } from "../../../utils/feltService";
+
+type AutoRenewalModalProps = {
+  handleClose: () => void;
+  isModalOpen: boolean;
+  callDataEncodedDomain: (number | string)[];
+  identity?: Identity;
+  domain?: string;
+  allowance?: string;
+};
+
+const AutoRenewalModal: FunctionComponent<AutoRenewalModalProps> = ({
+  handleClose,
+  isModalOpen,
+  callDataEncodedDomain,
+  identity,
+  domain,
+  allowance,
+}) => {
+  const { address } = useAccount();
+  const [price, setPrice] = useState<string>("0");
+  const [renewalAllowance, setRenewalAllowance] = useState<string>(
+    allowance ?? "0"
+  );
+  const [isTxSent, setIsTxSent] = useState(false);
+  const [isUsResident, setIsUsResident] = useState<boolean>(false);
+  const [usState, setUsState] = useState<string>("DE");
+  const [salesTaxRate, setSalesTaxRate] = useState<number>(0);
+  const [salesTaxAmount, setSalesTaxAmount] = useState<string>("0");
+  const [email, setEmail] = useState<string>("");
+  const [groups, setGroups] = useState<string[]>(["98125177486837731"]);
+  const [emailError, setEmailError] = useState<boolean>(true);
```

Comment:
In the modal that I tested You don't ask for email so I'm a bit confused

Reply:
I only ask for email and states if you have not already filled in the form (on registration for instance) and there are no results in the metahash database